app/models/component.rb

```diff
@@ -277,7 +277,8 @@ def duplicate(new_name: nil, new_prefix: nil, new_version: nil, new_release: nil
       copied_rule.fixtext = new_srg_rule.fixtext
 
       # Update associated tables (checks, disa_rule_descriptions) with new SRG rule data
-      copied_rule.disa_rule_descriptions = new_srg_rule.disa_rule_descriptions.map(&:dup)
+      copied_rule.disa_rule_descriptions.first.vuln_discussion =
+        new_srg_rule.disa_rule_descriptions.first.vuln_discussion
```

Comment:
Looking at the association `base_rule has_many disa_rule_descriptions`,
This will work if there's only one instance of disa_rule_description in the copied_rule and the new_srg_rule. Would that always be the case?
If not, what happens if the copied_rule and the new_srg_rule have more than one disa_rule_desc?
Or if for example the copied_rule has 2 disa_rule_descriptions and the new_srg_rule has 4.

Reply:
It should always be one and I'm not sure how it could ever have more than 1. There are other parts in the code as well using .first for these arrays of 1 I've found too.

Reply:
ok, should the association change for `has_many` to `has_one` ??

Reply:
Not as part of this PR at least. I think addressing that if needed is a much larger scope of work and I don't know all of the implications of it.